### PR TITLE
feat: implement todos

### PR DESCRIPTION
**Pull request by evilrobot-01**
- tackles various todos
- ports all the using tellor contract functional tests
- updates conversion of bytes to u256 to match using tellor contract implementation

### CODE REVIEW

File: src/impls.rs [hunk 1]
```diff
@@ -1490,20 +1487,131 @@ impl<T: Config> UsingTellor<AccountIdOf<T>> for Pallet<T> {
...
+			return None
+		}
+		timestamp_retrieved =
+			Self::get_timestamp_by_query_id_and_index(query_id, start).unwrap_or_default();
+		if timestamp_retrieved > timestamp {
+			// candidate found, check for disputes
+			search = false;
+		}
+		// since the value is within our boundaries, do a binary search
+		while search {
```
Comment (oraclown) [inline src/impls.rs:1513]: Could get rid of search variable and just use break statements?

**Review by oraclown — APPROVED** [review]
Thanks, @evilrobot-01 ! Looks good. Left a small questions

**Review by pawanbisht62 — APPROVED** [review]
LGTM

File: src/impls.rs (hunk 1 above)
Comment (evilrobot-01) [reply src/impls.rs:1513]: Took a look, but after implementing and then following clippy suggestions it resulted in code that was harder to read so feel it is fine to keep as is, especially as it aligns closer to source contract.

**Pull request merged**